lib/router.js

```diff
@@ -334,14 +334,13 @@ Router.prototype.routes = Router.prototype.middleware = function () {
     var matchedLayers = matched.pathAndMethod
     var mostSpecificLayer = matchedLayers[matchedLayers.length - 1]
     ctx._matchedRoute = mostSpecificLayer.path;
+    ctx.params = mostSpecificLayer.params( mostSpecificLayer.captures(path) )
     if (mostSpecificLayer.name) {
       ctx._matchedRouteName = mostSpecificLayer.name;
     }
 
     layerChain = matchedLayers.reduce(function(memo, layer) {
       memo.push(function(ctx, next) {
-        ctx.captures = layer.captures(path, ctx.captures);
-        ctx.params = layer.params(path, ctx.captures, ctx.params);
```

Comment:
This line is needed in order to pass in `params` to the `Router.prototype.param` function.

```js
ctx.params = layer.params(layer.captures(path))
```

Reply:
Addressed in https://github.com/ZijianHe/koa-router/pull/508